radix-engine-profiling/src/rocks_db_metrics/tests/common.rs

```diff
@@ -0,0 +1,661 @@
+    data: &mut BTreeMap<usize, Vec<Duration>>,
+    percent: f32,
+) -> Vec<(f32, f32)> {
+    assert!(percent <= 100f32);
+    let mut output_values = Vec::new();
+    for (k, v) in data.iter_mut() {
+        v.sort();
+        let idx = (((v.len() - 1) as f32 * percent) / 100f32).round() as usize;
+        output_values.push((*k as f32, v[idx].as_micros() as f32));
+    }
+    output_values
+}
+
+pub fn discard_spikes(data: &mut BTreeMap<usize, Vec<Duration>>, delta_range: f32) {
+    for (_, v) in data.iter_mut() {
+        v.sort();
+        let center_idx = v.len() / 2;
+        let median = v[center_idx].as_micros() as f32;
+
+        v.retain(|&i| {
+            let value = i.as_micros() as f32;
+            if value > median {
+                value - median <= delta_range
+            } else {
+                median - value <= delta_range
+            }
+        });
+    }
+}
+
+pub fn generate_commit_data(
+    rng: &mut ThreadRng,
+    value_size: usize,
+) -> (
+    DbPartitionKey,
+    DbSortKey,
+    IndexMap<DbSortKey, DatabaseUpdate>,
+) {
+    let mut value_data: DbSubstateValue = vec![0u8; value_size];
+    rng.fill(value_data.as_mut_slice());
+    let value = DatabaseUpdate::Set(value_data);
+
+    let mut node_id_value = [0u8; NodeId::UUID_LENGTH];
+    rng.fill(&mut node_id_value);
+    let node_id = NodeId::new(EntityType::InternalKeyValueStore as u8, &node_id_value);
+    let partition_key = SpreadPrefixKeyMapper::to_db_partition_key(&node_id, PartitionNumber(0u8));
+
+    let mut substate_key_value = [0u8; NodeId::LENGTH];
+    rng.fill(&mut substate_key_value);
+    let sort_key =
+        SpreadPrefixKeyMapper::to_db_sort_key(&SubstateKey::Map(substate_key_value.into()));
+
+    let mut partition = PartitionUpdates::new();
+    partition.insert(sort_key.clone(), value);
+
+    (partition_key, sort_key, partition)
+}
+
+pub fn prepare_db<S: SubstateDatabase + CommittableSubstateDatabase>(
+    substate_db: &mut S,
+    min_size: usize,
+    max_size: usize,
+    step: usize,
+    writes_count: usize,
+    random_size: bool,
+) -> Vec<(DbPartitionKey, DbSortKey, usize)> {
+    let mut data_index_vector: Vec<(DbPartitionKey, DbSortKey, usize)> =
+        Vec::with_capacity(max_size);
+
+    print!(
+        "Preparing database ({}, {}, {}, {})...",
+        min_size, max_size, step, writes_count
+    );
+    std::io::stdout().flush().ok();
+    let mut rng = rand::thread_rng();
+
+    if random_size {
+        println!("");
+        let batch_size = writes_count / 100;
+        for i in 0..writes_count / batch_size {
+            let mut input_data = DatabaseUpdates::with_capacity(batch_size);
+            for _ in 0..batch_size {
+                print!("\rRound {}/{}", i + 1, writes_count / batch_size);
+                std::io::stdout().flush().ok();
+
+                let size = rng.gen_range(min_size..=max_size);
+
+                let (partition_key, sort_key, partition) = generate_commit_data(&mut rng, size);
```

Comment:
Because we're generating a random, unique partition key each time, we're creating lower JMTs with size 1?

Reply:
Refactored `prepare_db()`, now `partition_key` is generated once for min~max substate size and is multiplied by writes count.